### PR TITLE
14 maximize coverage of mysqlgo

### LINKED ISSUE #14: maximize-coverage-of-mysql.go
Create table-driven unit tests to cover all of out database-implementation's methods and cover all lines except trivial ones such as error handling.

- [x] Need to add test for OpenID().
- [x]  Need to add test for AddUser, GetUser, GetUserID().
- [ ] Suppress unnecessary test logs.
- [ ] Reduce overhead, and unnecessary repetitive operations.

### PR DESCRIPTION
**Pull request by Alishah634**
## Rationale
This PR resolves #14  
The testing system needed to be improved and revamped, so it can be table driven to allow for descriptive tailored test cases for each function of the database. 

## Usage
N/A

### Non-trivial Files
Code coverage for mysql.go is now from approx. 55% to 65.9%. The remaining percentage is due to error checks that do not need to be tested.

- mysql_test.go was changed to add table driven unit tests to the AddUser(), GetUser(), GetUserID(). Additionally, minor refactoring was performed for code consistency and creating streamlined production code.
- Added a hot fix to mysql.go, the fix was to quickly implement empty strings validation for username, email, passwords.

### Testing
- [x] ➕ Added units to existing tests
- [ ] 🐣 Started a new group of tests
- [ ] 🙅 New tests aren't needed
- [ ] 🙋 Need help to add tests

### Dependencies
N/A
### Documentation Changes
- [ ] 📜 README.md
- [ ] 💬 Added **necessary** comments to existing code
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## Issues and Bugs
1. Found out there is no validation or error checking on the backend for if a user enters an empty string for username, email, password. 
2. Additionally, there needs to be functionality to check whether the email added is valid.

## Possible Solutions
Future development will have the variables passed into the function already validated. This kind of validation is not in the scope of this PR, and will require another issue ticket to be opened to be resolved.

## Additional Notes
Small concern that while adding table driven unit tests, the code base does increase quite a bit and some Test functions may not require this level of complexity to execute the test cases. Table driven tests may unnecessarily complex for the test cases.
For other notes refer to the Issues and Bugs Section of this PR.

### CODE REVIEW

File: internal/database/mysql_test.go [hunk 1]
```diff
@@ -236,68 +263,98 @@ func TestDeleteUser(t *testing.T) {
...
+                    t.Errorf("Expected created_on to be set, but got zero value")
+                }
+                if !user.IsValid() {
+                    t.Errorf("Expected valid user, but got invalid user: %v", user)
+                }
+            } else {
+                if user != nil {
+                    t.Errorf("Expected no user, but got: %v", user)
+                }
+            }
```
Comment (CarlosACJ55) [inline internal/database/mysql_test.go:313]: ditto here, I feel like we could make this code much much shorter and less indented with the other pattern.

File: internal/database/mysql_test.go [hunk 2]
```diff
@@ -236,68 +263,98 @@ func TestDeleteUser(t *testing.T) {
...
+    }
+
+    testCases := []struct {
+        name     string
+        field    string
+        value    string
+        expected int
+    }{
+        {"GetUserIDByUsername1", "username", "user_id_user", 1},
+        {"GetUserIDByEmail1", "email", "user_id_user@example.com", 1},
```
Comment (CarlosACJ55) [inline internal/database/mysql_test.go:340]: GetUser1ByEmail

File: internal/database/mysql_test.go [hunk 3]
```diff
@@ -236,68 +263,98 @@ func TestDeleteUser(t *testing.T) {
...
+        {"GetUserIDByUsername2", "username", "user2_id_user2", 2},
```
Comment (CarlosACJ55) [inline internal/database/mysql_test.go:341]: GetUser2IDByUsername, etc... Basically, the number should be labeling which user, not just apended at the end of the test name for clarity. Also, I thought we have ben using all_small_letter_and_underscores_for_test_names, probably best to keep one format

**Review by CarlosACJ55 — CHANGES_REQUESTED** [review]

File: internal/database/mysql_test.go [hunk 4]
```diff
@@ -56,7 +55,6 @@ func setup(t *testing.T) (subject *MySQLDatabase, probe *sql.DB) {
 
 func teardown(t *testing.T, subject *MySQLDatabase, probe *sql.DB) {
 	t.Helper()
-
```
[the remark this answers is not in the document]
Comment (Alishah634) [reply internal/database/mysql_test.go:59]: Okay

File: internal/database/mysql_test.go (hunk 2 above)
Comment (Alishah634) [reply internal/database/mysql_test.go:341]: Will refactor the test Description

File: internal/database/mysql_test.go (hunk 3 above)
Comment (Alishah634) [reply internal/database/mysql_test.go:342]: All test descriptions are Camel Case, and test variables/entrie/data are in the all_small_letter_and_underscore format. Eg. test_pass for password is in that format.

**Review by seoyoungcho213 — APPROVED** [review]

File: internal/database/mysql_test.go [hunk 5]
```diff
@@ -147,19 +145,48 @@ func TestAddUser(t *testing.T) {
...
-	if err := probe.QueryRow("SELECT id FROM users WHERE username = ?", "test_user2").Scan(&id); err != nil || id != 2 {
-		t.Fatal("Failed to sequentially create user with id 2 -", err)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := subject.AddUser(tc.username, tc.email, tc.password)
+			var id int
+			if tc.expectedError {
+				if err == nil {
+					t.Fatalf("Expected error but got none for case: %s", tc.name)
+				}
```
[the remark this answers is not in the document]
Comment (Alishah634) [reply internal/database/mysql_test.go]: I have refactored and updated the code to eliminate the inconsistency.

File: internal/database/mysql_test.go (hunk 1 above)
Comment (Alishah634) [reply internal/database/mysql_test.go]: I refactored the code both ways and it saves 1-2 lines. The refactoring is negligible, and this approach is descriptive and flexible for adding future tests.

**Review by CarlosACJ55 — APPROVED** [review]
Perfect! Good stuff @Alishah634 .

**Pull request merged**